File: 0078-subsets/solution.cpp
class Solution {
public:
vector<vector<int>> answer = {};
vector<int> currentSeq ={};
void recur(int index , vector<int>& nums){
    if(index == nums.size()){
        answer.push_back(currentSeq);
        return;
    }
    currentSeq.push_back(nums[index]);
    recur(index + 1 , nums);
    currentSeq.pop_back();
    recur(index + 1 , nums);
}
    vector<vector<int>> subsets(vector<int>& nums) {
        recur(0,nums);
        return answer;
        
    }
};

File: 0078-subsets/solution.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    List<Integer> currentSet = new ArrayList<>();
    public void recur(int index , int [] nums){
            if(index == nums.length){
                ans.add(new ArrayList<>(currentSet));
                return;
            }
            currentSet.add(nums[index]);
            recur(index+1,nums);
            currentSet.remove(currentSet.size()-1);
            recur(index+1,nums);
    }

    public List<List<Integer>> subsets(int[] nums) {
        recur(0,nums);
        return ans;
        
    }
}
